src/providers/twitch/TwitchIrcServer.cpp

```diff
@@ -935,31 +940,55 @@ void TwitchIrcServer::onReadConnected(IrcConnection *connection)
 {
     (void)connection;
 
-    std::lock_guard lock(this->channelMutex);
+    std::vector<ChannelPtr> activeChannels;
+    {
+        std::lock_guard lock(this->channelMutex);
 
-    // join channels
-    for (auto &&weak : this->channels)
+        activeChannels.reserve(this->channels.size());
+        for (const auto &weak : this->channels)
+        {
+            if (auto channel = weak.lock())
+            {
+                activeChannels.push_back(channel);
+            }
+        }
+    }
+
+    // get the selected channels
+    std::set<Channel *> selected;
+    for (auto *window : getApp()->getWindows()->allWindows())
     {
-        if (auto channel = weak.lock())
+        auto *page = window->getNotebook().getSelectedPage();
+        if (!page)
+        {
+            continue;
+        }
+
+        for (auto *split : page->getSplits())
         {
-            this->joinBucket_->send(channel->getName());
+            selected.emplace(split->getChannel().get());
         }
     }
```

Comment:
I would prefer if `WindowManager` instead got a public function that returns a list of channel names that are visible (e.g. `getVisibleChannelNames`), and `TwitchIrcServer` uses those channel names to prioritize which channels to load